equalsverifier-core/src/test/java/nl/jqno/equalsverifier/internal/util/ValidationsTest.java

```diff
@@ -1,13 +1,118 @@
 package nl.jqno.equalsverifier.internal.util;
 
 import static nl.jqno.equalsverifier.internal.testhelpers.Util.coverThePrivateConstructor;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.containsString;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
 
 import org.junit.jupiter.api.Test;
 
-public class ValidationsTest {
+/** Tests for {@link Validations} */
+class ValidationsTest {
 
     @Test
-    public void coverTheConstructor() {
+    void coverTheConstructor() {
         coverThePrivateConstructor(Validations.class);
     }
+
+    /**
+     * {@link Validations#validateFieldTypeMatches(Class, String, Class)} should
+     * throw if the field type does not match the given type.
+     */
+    @Test
+    void validateFieldTypeMatches_ShouldFailOnWrongType() {
+        {
+            Exception exc = assertThrows(IllegalStateException.class,
+                    () -> Validations.validateFieldTypeMatches(TestContainer.class, "listField", HashSet.class),
+                    "Should not allow HashSet as a List");
+            assertThat(exc.getMessage(),
+                    allOf(containsString("should be of type List"), containsString("but are HashSet")));
+        }
+        {
```

Comment:
I've never seen this use of blocks before. Is this a JUnit 5 thing? Will it run the second block even if the first one fails?

Reply:
Glad you ask. :-)   
No it is simple Java syntax.   
While it makes not much sense to use them in production code, I found benefits in using it in test code.

The benefits are the scoped variables. `exc` is only scoped inside the brackets. Thus, you can reuse the same name in another bracket, without having to fear copy paste errors, because you may have used `exc1`, `exc2` etc. (did happen to me more than once ;-) )  

Further, it separates the cases visually.  

It is something I came across a few months ago and found suitable use case for this with certain test setups. I think tests are a perfect fit for this if you don't want to use a separate method for each and every check.  
i.e. The alternatives are: To use a separate method for each check (which is totally valid).  
Or use  @ParameterizedTest.  
Or to be careful with the var-naming, so you don't mix up vars by accident.

Unfortunately, it does not have any impact on the test execution.  
For the use case you mentioned, `assertAll()`is a suitable method. It executes all given assertions, even if some fail.

Reply:
Alright, changed it to assertAll.   
for example, running the tests against the old èquals`code, produces these test failures:  
![image](https://github.com/user-attachments/assets/5cb26ea3-103e-454e-abc9-7326a24664ba)